Tensile/Source/lib/include/Tensile/Distance.hpp

```diff
@@ -331,6 +331,58 @@ namespace Tensile
             }
         };
 
+        template <typename Key>
+        struct GridBasedDistance : public Distance<Key>
+        {
+            enum
+            {
+                HasIndex = false,
+                HasValue = false
+            };
+
+            static std::string Type()
+            {
+                return "GridBased";
+            }
+            virtual std::string type() const override
+            {
+                return Type();
+            }
+
+            inline double operator()(Key const& p1, Key const& p2) const
+            {
+                double distance = 0.0;
+
+                double M = p2[0];
+                double N = p2[1];
+                double K = p2[2];
```

Comment:
Eventually we'll need to figure out a way to not have this hardcoded, but for now this is fine.